amd/src/annoto.js

```diff
@@ -183,6 +196,112 @@ define([
+                horizontalAlign = (innerAlignPlayers.indexOf(params.playerType) !== -1) ? 'inner' : 'element_edge';
+            } else if (params.widgetOverlay === 'inner') {
+                horizontalAlign = 'inner';
+            }
+
+            var configSrc = {
+                clientId: params.clientId,
+                position: params.position,
+                features: {
+                    tabs: params.featureTab,
+                    cta: params.featureCTA,
+                },
+                width: {
+                    max: (horizontalAlign === 'inner') ? 320 : 360,
+                },
+                align: {
+                    vertical: params.alignVertical,
+                    horizontal: horizontalAlign,
+                },
+                ux: {
+                    ssoAuthRequestHandle: function() {
+                        window.location.replace(params.loginUrl);
+                    },
+                },
+                zIndex: params.zIndex ? params.zIndex : 100,
+                demoMode: params.demoMode,
+                rtl: params.rtl,
+                locale: params.locale,
+            };
+
+            Object.assign(config, configSrc);
+
+            var widget = config.widgets[0];
+            var playerConfig = widget.player;
+            var ux = config.ux || {};
+            config.ux = ux;
+
+            /* ux.ssoAuthRequestHandle = this.ssoAuthRequestHandle; */
+            playerConfig.mediaDetails = this.enrichMediaDetails.bind(this);
+        },
+
+        enrichMediaDetails: function(details) {
+            // The details contains MediaDetails the plugin has managed to obtain
+            // This hook gives a change to enrich the details, for example
+            // providing group information for private discussions per course/playlist
+            // https://github.com/Annoto/widget-api/blob/master/lib/media-details.d.ts#L6.
+            var params = this.params;
+            var mediaDetails = {
+                description: params.mediaDescription,
+                group: {
+                    id: params.mediaGroupId,
+                    type: 'playlist',
+                    title: params.mediaGroupTitle,
+                    privateThread: params.privateThread,
+                }
+            };
+            /*
+             * Annoto Kaltura plugin, already has some details about the media like title.
+             * But for moodle if the title and description we get from Moodle is the activity and present,
+             * we should override it, if it's embedded in places where there is no Moodle media title, don't change it.
+             * The group should always be set as done here:
+             * https://github.com/Annoto/moodle-local_annoto/blob/master/amd/src/annoto.js#L142
+             *
+             * retVal.group = { ... };
+             */
+
+            var retVal = Object.assign(details, mediaDetails);
```

Comment:
@ilyaDymshits Does Moodle include polyfill for `Oeject.assign` for IE11? If not, don't use it.